website/source/docs/secrets/kv/versioned-kv.html.md

```diff
@@ -0,0 +1,383 @@
+---
+layout: "docs"
+page_title: "KV - Secrets Engines"
+sidebar_current: "docs-secrets-kv-versioned"
+description: |-
+  The KV secrets engine can store arbitrary secrets.
+---
+
+# Versioned KV Secrets Engine
+
+The `kv` secrets engine is used to store arbitrary secrets within the
+configured physical storage for Vault.
+
+Key names must always be strings. If you write non-string values directly via
+the CLI, they will be converted into strings. However, you can preserve
+non-string values by writing the key/value pairs to Vault from a JSON file or
+using the HTTP API. 
+
+This secrets engine honors the distinction between the `create` and `update`
+capabilities inside ACL policies.
+
+## Setup
+
+Most secrets engines must be configured in advance before they can perform their
+functions. These steps are usually completed by an operator or configuration
+management tool.
+
+A versioned `kv` secrets engine can be enabled by:
+
+```
+$ vault secrets enable vkv
+```
+
+Additionally, the versioned `kv` secrets engine is enabled by default at the
+path `secret/`. It can be disabled, moved, or enabled multiple times at
+different paths. Each instance of the KV secrets engine is isolated and unique.
+
+## Upgrading from Non-Versioned
+
+An existing non-versioned kv can be easily upgraded to a versioned key/value
+store with the CLI command:
+
+```
+$ vault kv enable-versioning secret/
+Success! Tuned the secrets engine at: secret/
+```
+
+or via the API:
+
+```
+$ cat payload.json
+{
+  "options": "versioned=true"
+}
+
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request POST \
+    --data @payload.json \
+    http://127.0.0.1:8200/v1/sys/mounts/secret/tune
+```
+
+This will start an upgrade process to upgrade the existing key/value data to
+a versioned format. The mount will be inaccessible during this process. This
+process could take a long time, so plan accordingly.  
+
+## ACL Rules
+
+The versioned kv store uses a prefixed API, which is different from the
+non-versioned store. Before upgrading from a non-versioned kv the ACL rules
+should be changed. Also different paths in the versioned API can be ACL'ed
+differently.
+
+Writing and reading versions are prefixed with the `data/` path. This policy
+that worked for the non-versioned kv:
+
+```
+path "secret/dev/team-1/*" {
+  capabilities = ["create", "update", "read"]
+}
+```
+
+Should be changed to:
+
+```
+path "secret/data/dev/team-1/*" {
+  capabilities = ["create", "update", "read"]
+}
+```
+
+There are different levels of data deletion for this backend. To grant a policy
+the permissions to delete the latest version of a key:
+
+```
+path "secret/data/dev/team-1/*" {
+  capabilities = ["delete"]
+}
+```
+To allow the policy to delete any version of a key:
+
+```
+path "secret/delete/dev/team-1/*" {
+  capabilities = ["create"]
```

Comment:
This should be "update"

Reply:
(I realize it *can* accept both, the way it's coded, but it's not really canonical to use "create". It might even be worth removing the CreateOperation here as it's a bit non-canonical. But it's probably not too important.)